executor/aggregate.go

```diff
@@ -95,7 +96,8 @@ func (e *HashAggExec) Next(goCtx goctx.Context) (Row, error) {
 }
 
 func (e *HashAggExec) getGroupKey(row Row) ([]byte, error) {
-	vals := make([]types.Datum, 0, len(e.GroupByItems))
+	var tmp [32]types.Datum
+	vals := tmp[:]
```

Comment:
The `vals` has len 32, you want `vals := tmp[:0]`.
And I think `make([]types.Datum, 0, 32)` is better, the memory is on stack too.